drivers/CoreLED/tests/CoreLED_test.cpp

```diff
@@ -0,0 +1,57 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "CoreLED.h"
+
+#include "gtest/gtest.h"
+#include "mocks/leka/SPI.h"
+
+using namespace leka;
+
+class CoreLEDTests : public ::testing::Test
+{
+  protected:
+	CoreLEDTests() : coreled(spimock) {}
+
+	void SetUp() override { expected_belt_color = RGB::pure_green; }
+	// void TearDown() override {}
+
+	CoreLED coreled;
+	SPIMock spimock;
+
+	RGB expected_belt_color;
+	RGB expected_belt_color_when_off;
+};
+
+TEST_F(CoreLEDTests, initialisation)
+{
+	ASSERT_NE(&coreled, nullptr);
+}
+
+TEST_F(CoreLEDTests, turnBeltOff)
+{
+	coreled.turnOff();
+
+	ASSERT_EQ(false, coreled.isOn());
+}
+
+TEST_F(CoreLEDTests, turnBeltOn)
+{
+	coreled.turnOn();
+
+	ASSERT_EQ(true, coreled.isOn());
+}
+
+TEST_F(CoreLEDTests, setImplementedBeltColor)
+{
+	coreled.setColor(expected_belt_color);
+	ASSERT_EQ(expected_belt_color, coreled.showColor());
+}
```

Comment:
Pour commencer simple, tu peux attendre que le spi soit appelé une fois pour transmettre la data

```suggestion
TEST_F(CoreLEDTests, setImplementedBeltColor)
{
	EXPECT_CALL(spimock, write).Times(1);
	coreled.setColor(expected_belt_color);
}
```

--

Comme ici on souhaite vérifier que la couleur a bien été envoyé et sous le bon format en SPI, on va regarder l'array qui va transiter. La vérification des arguments est simple, et dans le cas d'un array il va falloir s'armer d'autres outils gtest, ci-dessous un exemple avec `ElementsAre`

```suggestion
using ::testing::ElementsAre;

...

TEST_F(CoreLEDTests, setImplementedBeltColor)
{
	auto expected_array = ElementsAre(1, 1, 2, 3, 5);
	EXPECT_CALL(spimock, write(expected_array)).Times(1);
	coreled.setColor(expected_belt_color);
	// Il faut bien sûr faire le lien entre expected_belt_color et expected_array au préalable
}
```

Il existe bien d'autres manières de le faire, par ici c'est un bon point d'entrée: [[Link](https://github.com/google/googletest/blob/main/docs/gmock_cook_book.md#verifying-complex-arguments-saveargverify)]